trl/trainer/dpo_trainer.py

```diff
@@ -1389,20 +1392,23 @@ def evaluation_loop(
 
             policy_output_decoded, ref_output_decoded = self.generate_from_model_and_ref(self.model, random_batch)
 
-            self.log(
-                {
-                    "game_log": wandb.Table(
-                        columns=["Prompt", "Policy", "Ref Model"],
-                        rows=[
-                            [prompt, pol[len(prompt) :], ref[len(prompt) :]]
-                            for prompt, pol, ref in zip(
-                                random_batch["prompt"], policy_output_decoded, ref_output_decoded
-                            )
-                        ],
+            table = pd.DataFrame(
+                columns=["Prompt", "Policy", "Ref Model"],
+                data=[
+                    [prompt, pol[len(prompt) :], ref[len(prompt) :]]
+                    for prompt, pol, ref in zip(
+                        random_batch_dataset["prompt"], policy_output_decoded, ref_output_decoded
                     )
-                }
+                ],
             )
-            self.state.log_history.pop()
+            if "wandb" in self.args.report_to:
+                wandb.log({"game_log": wandb.Table(data=table)})
+
+            if "comet_ml" in self.args.report_to:
+                log_table_to_comet_experiment(
+                    name="game_log.csv",
```

Comment:
can you save it in the `output_dir` instead?

Reply:
Comet SDK post submitted files to the server in the background. The intermediate copy of the file is stored in the temporary directory until upload to the Comet server completes.

Could you please elaborate on your idea so I can understand it better?

Reply:
When you run `experiment.log_table(tabular_data=table, filename=name)`, does it save locally something in a filename `name`?

Reply:
It is saves indeed the table data into temporary file in temporary directory. This file lives until its upload to the Comet server is complete. After that it is automatically cleaned either by Comet SDK or by OS if something goes bad during Python script execution.